File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        //1. Output list node
        ListNode dummy = new ListNode(-1);
        ListNode resultant = dummy;
        int carry = 0;
        int sum = 0;
        
        //2. While loop
        while(l1!=null || l2!=null||carry!=0){
            //a) initialize the values
            int firstVal = (l1!=null)?l1.val:0;
            int secondVal = (l2!=null)?l2.val:0;
            
            //b) inialize the sum 
            sum = firstVal+ secondVal+carry;
            carry = sum/10;
            resultant.next = new ListNode(sum%10);
            resultant = resultant.next;
            
            
            //c.) go to the next values
            l1 = (l1!=null)?l1.next:null;
            l2 = (l2!=null)?l2.next:null;
            
        }
        return dummy.next;
        
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1)
        current = dummy 
        
        #1. Loop through everything
        carry = 0
        while(l1 or l2 or carry!=0 ):
            l1Val = l1.val if l1 else 0
            l2Val = l2.val if l2 else 0
            
            totalSum = l1Val+ l2Val+ carry
            carry = totalSum//10
            
            temp = totalSum%10
            tempNode = ListNode(temp)
            current.next = tempNode
            current = current.next
            
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        
        return dummy.next